File: DemoServer/AdditionalLanguages/python/staticIndexes/fanoutIndex/fanoutIndex.py
from typing import List
from demo_example import Example, RunParamsBase
from models import Order

#region Usings
from ravendb import AbstractIndexCreationTask
#endregion

class RunParams(RunParamsBase):
    def __init__(self, name_prefix: str = None):
        self.name_prefix = name_prefix

#region Demo
#region Step_1
class Orders_ByProductDetails(AbstractIndexCreationTask):
#endregion

    #region Step_2
    class IndexEntry:
        def __init__(self, product_id: str, product_name: str):
            self.product_id = product_id
            self.product_name = product_name
    #endregion

    #region Step_3
    def __init__(self):
        super().__init__()
        self.map = (
            "docs.Orders.SelectMany(order => order.Lines, (order, orderLine) => new {"
            "    product_id = orderLine.Product,"
            "    product_name = orderLine.ProductName"
            "})"
        )
    #endregion
#endregion

class FanoutIndex(Example):
    def run(self, run_params: RunParams) -> List[Order]:
        namePrefix = run_params.name_prefix or "Chocolade"
        Orders_ByProductDetails().execute(self.document_store_holder.store())
        #region Demo
        #region Step_4
        with self.document_store_holder.store().open_session() as session:
            orders = list(
                session.query_index_type(Orders_ByProductDetails, Orders_ByProductDetails.IndexEntry)
                    .where_starts_with("product_name", namePrefix)
            )
        #endregion
        #endregion

        return orders

File: DemoServer/AdditionalLanguages/php/staticIndexes/fanoutIndex/FanoutIndex.php
<?php

namespace RavenDB\Demo\staticIndexes\fanoutIndex;

//region Usings
use RavenDB\Documents\Indexes\AbstractIndexCreationTask;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Order;

//region Demo
//region Step_1
class Orders_ByProductDetails extends AbstractIndexCreationTask
//endregion
{
    //region Step_2
    public function __construct()
    {
        parent::__construct();
            $this->map = "docs.Orders.SelectMany(order => order.Lines, (order, orderLine) => new {" .
                "    ProductId = orderLine.Product," .
                "    ProductName = orderLine.ProductName" .
                "})";
        }
    //endregion
}
//endregion

//region Demo
//region Step_3
class IndexEntry
{
    private ?string $productId = null;
    private ?string $productName = null;

    public function getProductId(): ?string
    {
        return $this->productId;
    }

    public function setProductId(?string $productId): void
    {
        $this->productId = $productId;
    }

    public function getProductName(): ?string
    {
        return $this->productName;
    }

    public function setProductName(?string $productName): void
    {
        $this->productName = $productName;
    }
}
//endregion
//endregion

class FanoutIndex
{
    public function __invoke(RunParams $runParams): array
    {
        $namePrefix = $runParams->getNamePrefix() ?? "Chocolade";

        //region Demo
        $orders = [];

        //region Step_4
        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {

            $orders = $session->query(Order::class, Orders_ByProductDetails::class)
                ->whereStartsWith("ProductName", $namePrefix)
                ->toList();
                
        } finally {
            $session->close();
        }
        //endregion
        //endregion

        return $orders;
    }
}
